### PR TITLE
Minor improvements to the Optionals documentation

### PR DESCRIPTION
**Pull request by g-cassie**
- Document how to do a while loop with both an optional capture and a continue expression
- Add links to end of Optionals section to improve discoverability of optional capture behaviour in if and while statements.

Based on my own experience having difficulty finding these things.

### CODE REVIEW

File: doc/langref.html.in [hunk 1]
```diff
@@ -4635,6 +4635,14 @@ test "while null capture" {
     } else {
         try expect(sum2 == 3);
     }
+
```
Comment (rootbeer) [inline doc/langref.html.in:4638]: Add a short comment here to highlight what's going on?  Something like:

    // null capture with a continue expression 

(Alternatively, you could put this comment in the name of a new test block (similar to "while loop continue expression, more complicated" above)

**Review by rootbeer — COMMENTED** [review]
:+1:

File: doc/langref.html.in [hunk 2]
```diff
@@ -4915,87 +4925,92 @@ test "if boolean" {
...
-    } else |_| {
-        unreachable;
+        value.* = 2;
     }
 
     if (c) |value| {
-        try expect(value == 9);
-    } else |_| {
+        try expect(value == 2);
+    } else {
```
Comment (g-cassie) [inline doc/langref.html.in:5013]: The diff is terrible but basically I took the two sections relating to optional and put them under their own heading.

Comment (rootbeer) [reply doc/langref.html.in:5013]: Oof, yeah the diff is not helping.   The new if-err-capture examples all look reasonable to me.

**Review by rootbeer — COMMENTED** [review]
I think the while-else examples are a bit contrived, but they get the point across without being complicated.  I think you'd need to add a `break` (?) into the body to see the impact of the else?  (Ah, this has come up before, see https://github.com/ziglang/zig/issues/2614#issuecomment-500216492).

Fixes all LGTM.

File: doc/langref.html.in (hunk 2 above)
Comment (g-cassie) [reply doc/langref.html.in:5013]: I didn't add anything - it is just exactly as it was but split into two sections.  I built the docs and they work as expected locally.

**Pull request merged**